avocado/utils/network.py

```diff
@@ -71,3 +73,44 @@ def find_free_ports(start_port, end_port, count, address="localhost"):
             count -= 1
         i += 1
     return ports
+
+
+class PortTracker(Borg):
+
+    """
+    Tracks ports used in the host machine.
+    """
+
+    def __init__(self):
+        Borg.__init__(self)
+        self.address = 'localhost'
+        self.start_port = 5000
+        if not hasattr(self, 'retained_ports'):
+            self._reset_retained_ports()
+
+    def __str__(self):
+        return 'Ports tracked: %r' % self.retained_ports
+
+    def _reset_retained_ports(self):
+        self.retained_ports = []
+
+    def register_port(self, port):
```

Comment:
I think it'd be really useful to have `force` here to force inject it even on use (eg. when you start service, obtain address from that service and then keep track of it).